app/src/main/res/layout/activity_main.xml

```diff
@@ -7,12 +7,23 @@
     tools:context=".MainActivity">
 
     <TextView
+        android:id="@+id/welcome_text_field"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
+        android:text="@string/welcome_text_field_text"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toTopOf="@+id/check_button"
```

Comment:
Be careful with this, if you define an id with a + like here, you are creating a new reference. That should be used only when we are defining an id.

please remove that

Reply:
good to know! done